nightfall-optimist/src/event-handlers/transaction-submitted.mjs

```diff
@@ -68,15 +68,20 @@ async function transactionSubmittedEventHandler(eventParams) {
     transaction = await checkAlreadyInBlock(transaction);
     // save transaction if not in block
     if (fromBlockProposer) {
-      saveTransaction({ ...transaction });
+      saveTransaction({ ...transaction }).catch(err =>
+        logger.error({ msg: 'Error while saving transaction', err }),
+      );
     }
 
     await checkTransaction(transaction, true);
+
     logger.info('Transaction checks passed');
 
     // save it
     if (!fromBlockProposer) {
-      saveTransaction({ ...transaction });
+      saveTransaction({ ...transaction }).catch(err =>
+        logger.error({ msg: 'Error while saving transaction', err }),
+      );
```

Comment:
Hey, sorry @imagobea, I just saw your comment now. Yes, I find it confusing as well, but I didn't take much attention and I was trying to avoid touching things that work.